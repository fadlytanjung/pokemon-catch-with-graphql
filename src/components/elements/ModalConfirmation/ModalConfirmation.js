import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';
import Typography from '../Typography';
import Modal from '../Modal';
import styles from './styles.scoped.css';

export default function ModalConfirmation(props) {
  const { buttonCancel, buttonConfirm, children, onCancel, open, onConfirm, title, subtitle } = props;
  return (
    <Modal className={styles.root} onCancel={onCancel} open={open}>
      <Typography tag="h1" variant="headline-medium" weight="bold">{title}</Typography>
      <Typography tag="p" variant="caption">{subtitle}</Typography>
      {children}
      <footer>
        {onConfirm &&
          <Button onClick={onConfirm}>
            {onConfirm ? buttonConfirm : buttonCancel}
          </Button>
        }
        <Button onClick={onCancel} variant="outline">{buttonCancel}</Button>
      </footer>
    </Modal>
  );
}

ModalConfirmation.defaultProps = {
  buttonCancel: 'Close',
  buttonConfirm: 'Confirm',
  children: undefined,
  onCancel: undefined,
  onConfirm: undefined,
  open: false,
  reverseBtn: false,
  subtitle: '',
  title: '',
};

ModalConfirmation.propTypes = {
  buttonCancel: PropTypes.string,
  buttonConfirm: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.string, PropTypes.object, PropTypes.any]),
  onCancel: PropTypes.func,
  onConfirm: PropTypes.oneOfType([PropTypes.func, PropTypes.any]),
  open: PropTypes.bool,
  reverseBtn: PropTypes.bool,
  subtitle: PropTypes.string,
  title: PropTypes.string,
};
