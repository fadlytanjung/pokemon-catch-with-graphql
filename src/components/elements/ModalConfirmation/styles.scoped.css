.root > header > img:first-child {
  height: 100px;
}

.root > h1 {
  /* font-size: 1.75rem; */
  /* font-weight: 800; */
  line-height: 1.2;
  margin-bottom: 12px;
  text-align: center;
}

.root > p {
  /* color: #000; */
  line-height: 1.6;
  text-align: center;
}

.root > footer {
  display: flex;
  margin-top: 32px;
}

.root > footer > a {
  flex: 1;
}
.root > footer > button {
  flex: 1;
}
.root > footer > button:first-child {
  margin-right: 0.75rem;
}

.root > footer > a:first-child {
  margin-right: 1.25rem;
}

@media only screen and (max-width: 48em) {

}
