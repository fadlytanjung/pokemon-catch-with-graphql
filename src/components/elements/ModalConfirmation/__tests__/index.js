import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ModalConfirmation from '../ModalConfirmation';

describe('src/components/elements/ModalConfirmation', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ModalConfirmation />);
    expect(tree).toMatchSnapshot();
  });
});
