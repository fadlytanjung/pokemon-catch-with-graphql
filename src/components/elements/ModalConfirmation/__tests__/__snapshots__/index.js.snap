// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/ModalConfirmation render 1`] = `
<Modal
  className=""
  onClose={[Function]}
  open={false}
>
  <Typography
    tag="h1"
    variant="headline-medium"
    weight="bold"
  >
    
  </Typography>
  <Typography
    tag="p"
    variant="caption"
    weight="reguler"
  >
    
  </Typography>
  <footer>
    <Button
      className=""
      disabled={false}
      height={48}
      onClick={null}
      to=""
      type="button"
      variant="outline"
      width={168}
    >
      Close
    </Button>
  </footer>
</Modal>
`;
